src/tools/translate-to-fuzz.h

```diff
@@ -0,0 +1,1106 @@
+  static const int USABLE_MEMORY = 32;
+
+  // the number of runtime iterations (function calls, loop backbranches) we
+  // allow before we stop execution with a trap, to prevent hangs. 0 means
+  // no hang protection.
+  static const int HANG_LIMIT = 25;
+
+  // Optionally remove NaNs, which are a source of nondeterminism (which makes
+  // cross-VM comparisons harder)
+  static const bool DE_NAN = true;
+
+  // after we finish the input, we start going through it again, but xoring
+  // so it's not identical
+  int xorFactor = 0;
+
+  int8_t get() {
+    if (pos == bytes.size()) {
+      // we ran out of input, go to the start for more stuff
+      finishedInput = true;
+      pos = 0;
+      xorFactor++;
+    }
+    return bytes[pos++] ^ xorFactor;
+  }
+
+  int16_t get16() {
+    auto temp = uint16_t(get()) << 8;
+    return temp | uint16_t(get());
+  }
+
+  int32_t get32() {
+    auto temp = uint32_t(get16()) << 16;
+    return temp | uint32_t(get16());
+  }
+
+  int64_t get64() {
+    auto temp = uint64_t(get32()) << 32;
+    return temp | uint64_t(get32());
+  }
+
+  float getFloat() {
+    return Literal(get32()).reinterpretf32();
+  }
+
+  double getDouble() {
+    return Literal(get64()).reinterpretf64();
+  }
+
+  void build() {
+    setupMemory();
+    // keep adding functions until we run out of input
+    while (!finishedInput) {
+      addFunction();
+    }
+    if (HANG_LIMIT > 0) {
+      addHangLimitSupport();
+    }
+    if (DE_NAN) {
+      addDeNaNSupport();
+    }
+  }
+
+  void setupMemory() {
+    wasm.memory.exists = true;
+    // use one page
+    wasm.memory.initial = wasm.memory.max = 1;
+  }
+
+  const Name HANG_LIMIT_GLOBAL = "hangLimit";
+
+  void addHangLimitSupport() {
+    auto* glob = new Global;
+    glob->name = HANG_LIMIT_GLOBAL;
+    glob->type = i32;
+    glob->init = builder.makeConst(Literal(int32_t(HANG_LIMIT)));
+    glob->mutable_ = true;
+    wasm.addGlobal(glob);
+
+    auto* func = new Function;
+    func->name = "hangLimitInitializer";
+    func->result = none;
+    func->body = builder.makeSetGlobal(glob->name,
+      builder.makeConst(Literal(int32_t(HANG_LIMIT)))
+    );
+    wasm.addFunction(func);
+
+    auto* export_ = new Export;
+    export_->name = func->name;
+    export_->value = func->name;
+    export_->kind = ExternalKind::Function;
+    wasm.addExport(export_);
+  }
+
+  Expression* makeHangLimitCheck() {
+    return builder.makeSequence(
+      builder.makeIf(
+        builder.makeUnary(
+          UnaryOp::EqZInt32,
+          builder.makeGetGlobal(HANG_LIMIT_GLOBAL, i32)
+        ),
+        makeTrivial(unreachable)
+      ),
+      builder.makeSetGlobal(
+        HANG_LIMIT_GLOBAL,
+        builder.makeBinary(
+          BinaryOp::SubInt32,
+          builder.makeGetGlobal(HANG_LIMIT_GLOBAL, i32),
+          builder.makeConst(Literal(int32_t(1)))
+        )
+      )
+    );
+  }
+
+  void addDeNaNSupport() {
```

Comment:
The alternating caps here kinda hurts my eyes. `addDeNanSupport`?

Alternatively, `addNanRemovalSupport`, and change "de-NaN" to "remove NaN" throughout.
I think just changing `addDeNaNSupport` to `addRemoveNaNSupport` is less jarring. Specifically the high-low alternating character pattern is what I think is most problematic (e.g. "QqQqQ").